wp-modules/pattern-data-handlers/pattern-data-handlers.php

```diff
@@ -480,6 +455,16 @@ function move_block_images_to_theme( $pattern_html ) {
 			)
 		);
 
+		// Remove any URL query vars from the filename.
+		$url_parts = explode( '?', $url_found );
+		$filename  = basename( $url_parts[0] );
+
+		if ( 0 === strpos( $url_found, get_stylesheet_directory_uri() . '/patterns/images' ) ) {
+			// The image is already in the theme.
+			$pattern_html = str_replace( $url_found, "<?php echo esc_url( get_stylesheet_directory_uri() ); ?>/patterns/images/$filename", $pattern_html );
```

Comment:
If the image URL in `<theme name>/patterns/images/`, it shouldn't be necessary to do `wp_remote_retrieve_response_code()` below to check if the image actually exists. 

This can do the string replace below with the PHP tag, and continue.

If it doesn't exist, the URL will fail either way.